websites/R/RadioMe/presence.ts

```diff
@@ -1,220 +1,131 @@
-        } else {
-          //Radio is stopped
-          browsingTimestamp = 0;
-
-          presenceData.smallImageKey = "pause";
-          presenceData.smallImageText = (await strings).pause;
-
-          presenceData.details =
-            document.getElementsByClassName("song-name")[0].textContent;
-        }
-      } else {
-        //Player is inactive
-
-        presence.setActivity();
-        return;
-      }
+      handleUnknown(path);
       break;
   }
+});
+
+async function getStrings() {
+  return presence.getStrings(
+    {
+      play: "general.playing",
+      pause: "general.paused",
+      search: "general.searching",
+      browsing: "general.browsing"
+    },
+    await presence.getSetting<string>("lang").catch(() => "en")
+  );
+}
+
+/**
+ * Handle radio station
+ */
+function handleStation(): void {
+  const station =
+    document.querySelector<HTMLSpanElement>(".song-name")?.textContent;
+
+  let presenceData: PresenceData;
+
+  // Check if the playing icon is shown
+  if (document.querySelector<HTMLDivElement>(".playbutton-global-playing")) {
+    // Radio is playing / buffering
+    timestamp &&= Date.now();
+
+    presenceData = {
+      details: station,
+      largeImageKey: (
+        document.querySelector<HTMLAnchorElement>("#player-station-logo-link")
+          .children[0] as HTMLImageElement
+      ).src,
+      smallImageText: strings.play,
+      smallImageKey: "play",
+      startTimestamp: timestamp
+    };
+  } else {
+    // Radio is paused
+    timestamp = 0;
+
+    presenceData = {
+      details: station,
+      largeImageKey: (
+        document.querySelector<HTMLAnchorElement>("#player-station-logo-link")
+          .children[0] as HTMLImageElement
+      ).src,
+      smallImageText: strings.pause,
+      smallImageKey: "pause"
+    };
+  }
 
   presence.setActivity(presenceData);
-});
+}
+
+/**
+ * Handle search
+ */
+function handleSearch(): void {
+  const presenceData: PresenceData = {
+    details: new URLSearchParams(window.location.search).get("term"),
+    state: document.querySelector<HTMLHeadingElement>("h1").textContent,
+    largeImageKey: "logo_big",
+    smallImageText: strings.search,
+    smallImageKey: "search"
+  };
+
+  presence.setActivity(presenceData);
+}
+
+/**
+ * Handle generic pages
+ * @param preferTitle Prefer document title over h1 text content
+ */
+function handleGeneric(preferTitle = false): void {
+  presence.setActivity({
+    details: preferTitle
+      ? document.title
+      : document.querySelector<HTMLHeadingElement>("h1")?.textContent ??
+        document.title,
+    largeImageKey: "logo_big"
+  });
+}
+
+/**
+ * Handle unknown page
+ * @param path Page path
+ */
+function handleUnknown(path: string[]): void {
+  const region = [
+    ...document.querySelectorAll<HTMLAnchorElement>(".region-btn")
+  ]
+    .filter(e => e.classList.contains("active"))[0]
+    .pathname?.slice(1);
+
+  if (region !== "" && path[0] === region) {
```

Comment:
You don’t get it do ya
```suggestion
  if (region && path[0] === region) {
```

Reply:
An example would have been useful the first time...
I'm going to change it.

Reply:
You should know that empty strings are falsy

Reply:
Sure but I didn't see what you ment directly. Sorry but after more than 8 hours working on these presences explaining various decisions it's hard to calm down and think straight.